server/handlers/postSurveys.js

```diff
@@ -0,0 +1,44 @@
+const { getDB } = require('../databaseConnection');
+const addQuestions = require('../queries/addQuestions');
+const createSurvey = require('../queries/createSurvey');
+/* Survey schema:
+    {
+        id: ObjectID (auto generated)
+        title: string
+        description: string
+        status: created/published/closed
+        dateCreated: date
+        dateToPublish: date
+        datePublished: date
+        dateToClose: date
+        dateClosed: date
+        anonymous: boolean
+        recipients: [{employeeId: ObjectId completed: boolean}]
+        questions: [{questionId: ObjectId position: int}]
+        responses: [{
+            employeeId: ObjectID (or null if anon.)
+            answers: [{
+                questionId: ObjectId
+                answer: string/int (depends on question type)
+                comment: null or string
+                }]
+        }]
+    }
+*/
+
+const postSurveys = async (req, res) => {
+  const surveyObj = req.body;
```

Comment:
Input validation will be needed to ensure the req body matches some schema!
https://hapi.dev/family/joi/?v=16.1.8

Reply:
+1